core-processor/src/ext.rs

```diff
@@ -1211,6 +1248,38 @@ mod tests {
         assert_eq!(0, allowance);
     }
 
+    #[test]
+    fn test_send_limit() {
+        // Max of 3 sends
+        let mut ext = Ext::new(ProcessorContextBuilder::new()
+            .with_max_outgoing_messages(3)
+            .build());
+
+        let handle = ext.send_init();
+        assert!(handle.is_ok());
+
+        // Default data
+        let data = HandlePacket::new(ProgramId::default(), LimitedVec::default(), 0);
+
+        let msg = ext.send_commit(handle.unwrap(), data, 0);
+        assert!(msg.is_ok());
+
+        let handle = ext.send_init();
+        assert!(handle.is_ok());
+
+        let msg = ext.send_push(handle.unwrap(), &[]);
+        assert!(msg.is_ok());
+
+        let handle = ext.send_init();
+        assert!(handle.is_ok());
+
+        let msg = ext.send_push_input(handle.unwrap(), 0, 0);
```

Comment:
assert is ok?

Reply:
Replaced with unwrapping, better aswell since it will show what error happened

Reply:
> Replaced with unwrapping, better aswell since it will show what error happened

It's better to avoid unwraps even for tests
Expect does the same but can provide some (even short) explanation for other devs what went wrong within the test.